### PR TITLE
[OP] Add softmax_cross_entropy loss

### PR DESCRIPTION
**Pull request by tqchen**
cc @pluskid @antinucleon

### CODE REVIEW

File: src/operator/loss_binary_op-inl.h [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+namespace mxnet {
+namespace op {
+
+// return a shape of scalar
+inline TShape SoftmaxCrossEntropyShape_(const TShape& lshape,
+                                        const TShape& rshape,
+                                        const EnvArguments& env) {
+  mshadow::index_t shape[] = {1};
+  CHECK_EQ(lshape.ndim(), 2)
+      << "SoftmaxCrossEntroy only accept 2D data";
```
Comment (vchuravy) [inline src/operator/loss_binary_op-inl.h]: type `Entrop` should be `Entropy` also the following lines.

**Comment by vchuravy** [comment]
What is the difference between `softmax_cross_entropy_loss` and `SoftmaxOutput`?

- `SoftmaxOutput` is softmax combined with logloss (average cross entropy loss) for backwards
- `SoftmaxActivation` is applying only softmax forward&backwards
- `SoftmaxCrossEntropy` is ???

**Comment by tqchen** [comment]
in forward, softmax_cross_entropy output the cross entropy value, while SoftmaxOutput directly outputs the probability.

**Comment by tqchen** [comment]
To be clear, this is the correct formal definition of softmax_cross_entropy loss. It also enables you to further composite softmax loss function with other losses like sum them together. However, the SoftmaxOutput still serve as a handy tool to directly being used as terminal node where you need the output result
